✨ Improve Google matched content logic to reflect adsbygoogle.js.

nbeloglazov opened the pull request:
The commit introduces core-shared.js file which supposed to be shared between AMP and internal Google adsbygoogle.js codebase. The file has minimal dependencies on environment to make keeping them in sync easier.

In particular this commit adds support for publisher settings as described in https://support.google.com/adsense/answer/7533385

Tested:
  unit tests
  manually on fast fetch and delayed fetch

torch2424 commented on ads/google/core-shared.js:
```diff
@@ -0,0 +1,482 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
```
Incorrect copyright year 😄

nbeloglazov replied:
Thanks, done.

lannka commented on ads/google/core-shared.js:
```diff
@@ -0,0 +1,482 @@
+ * @param {!LayoutType} layout
+ * @return {number}
+ */
+const getAdHeight = (adWidth, layout) => {
```
consistency: shall we use `function`?

nbeloglazov replied:
Done.

lannka commented on ads/google/core-shared.js:
```diff
@@ -0,0 +1,482 @@
+ * will be used in ad request.
+ * @record
+ */
+class CoReConfig { // eslint-disable-line no-unused-vars
```
Does a TypeDef work here? Is a class needed?

nbeloglazov replied:
Generally styleguide recommends records over typedefs: https://google.github.io/styleguide/jsguide.html#jsdoc-enum-and-typedef-comments

lannka commented on ads/google/core-shared.js:
```diff
@@ -0,0 +1,482 @@
+ * @param {boolean} isMobile
+ * @return {!CoReConfig}
+ */
+export function getAutoConfig(availableWidth, isMobile) {
```
please organize the file so that public function is before local function.

nbeloglazov replied:
Done.

lannka commented on ads/google/core-shared.js:
```diff
@@ -0,0 +1,482 @@
+ * than this will be adapted to the mobile logic.
+ * @const {number}
+ */
+export const MIN_PUB_CONTROL_WIDTH_OF_DESKTOP = 468;
```
please move public constants before local ones

nbeloglazov replied:
Done.

nbeloglazov commented:
Addressed comments. And refactored code to not depend on any other AMP libraries as the code will be imported and used by internal google tags. Please take another look. Thanks!

keithwrightbos commented on ads/google/core-shared.js:
```diff
@@ -0,0 +1,520 @@
+ * Get CoRe Pub Control UI Sizes.
+ * @param {number} availableWidth
+ * @param {{
+ *   numberOfRows: (string|undefined),
```
You repeat this type multiple times, can you make it a proper typedef?

nbeloglazov replied:
Done. Introduced RawPublisherControlParams typedef.
